pkg/sign/internal/kmssigner/kmssigner.go

```diff
@@ -0,0 +1,186 @@
+// +skip_license_check
+
+// Copyright © 2018 Heptio
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package kmssigner implements a crypto.Signer backed by Google Cloud KMS.
+package kmssigner
+
+// This has been modified to suit cert-manager's use case; we add support for SHA512
+// digests, which are forced by helm. This is copied across to the cert-manager/release
+// project unless/until the changes are merged upstream in:
+// https://github.com/heptiolabs/google-kms-pgp/
```

Comment:
They've certainly not been responsive to my offers for [upstreaming](https://github.com/heptiolabs/google-kms-pgp/issues/6) these changes.

I don't really know what we'd gain from forking that we haven't already gained from just copying the file (which is sort of like forking, I guess?). Plus, I intentionally put the copied code into an internal package so we don't run the risk of someone importing it.

If we fork it, presumably that would be so that others can make use of it and at the end of the day this library might encourage the use of PGP and that would be actively harmful :joy:

Reply:
haha yeah that's a good reason to not fork ^, I'm sold :)